tests/test_run_rvic.py

```diff
@@ -1,19 +1,45 @@
 import pytest
 
-from osprey_flask_app import run_rvic
+from osprey_flask_app import create_app
 from wps_tools.testing import url_path
-from tempfile import NamedTemporaryFile
 from pkg_resources import resource_filename
 import os
-import requests
 
 
-def full_rvic_test(kwargs):
-    outpath = run_rvic.run_full_rvic(kwargs)
-    outpath_url = requests.get(outpath)
-    with NamedTemporaryFile(suffix=".nc", dir="/tmp") as outfile:
-        outfile.write(outpath_url.content)
-        assert os.path.isfile(outfile.name)
+@pytest.fixture
+def client():
+    flask_app = create_app()
+
+    # Create a test client using the Flask application configured for testing
+    with flask_app.test_client() as testing_client:
+        # Establish an application context
+        with flask_app.app_context():
+            yield testing_client
+
+
+def full_rvic_test(kwargs, client, valid_input=True):
```

Comment:
Generally I am in favour of factoring out common methods, but we are only using this code in one spot. We may want to consider moving this back under the params.